### ISSUE TITLE
my customer acting funny

### ISSUE THREAD

**Issue by heokieee** [nus-cs2030/2324-s2#774]
<img width="1470" alt="Screenshot 2024-04-27 at 00 49 41" src="https://github.com/nus-cs2030/2324-s2/assets/157129844/7050e5fe-0572-4151-bc6f-a31df7aa0e28">
hi i followed the test case for 14.in but failed miserably. Customer 18 queues at self check out 4 but it gets served by server 2. does anyone knows my mistake??

_Originally posted by @heokieee in https://github.com/nus-cs2030/2324-s2/issues/758#issuecomment-2079746083_

**Comment by heokieee** [comment]
<img width="640" alt="Screenshot 2024-04-27 at 02 31 02" src="https://github.com/nus-cs2030/2324-s2/assets/157129844/be84e293-b911-499a-80e5-940e0999098b">

this is my pendingServe part

**Comment by starryfieldz** [comment]
Yes i agree with @nicolenahh, when your event changes from Arrive to Wait, you will need to design it such that it can only wait at first SelfCheckout and not the other CheckOutServers. I think having queue at first Self-checkout or having queues at all self checkouts both works, what matters is the design such that customer can only wait at first SelfCheckout

**Comment by lohkeyyy** [comment]
the issue is not due to your customer but due to your SelfCheckOuts

**Comment by ZHD1987E** [comment]
finals is over you can close this issue
